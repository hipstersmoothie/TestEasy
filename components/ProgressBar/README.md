# @test-easy/progress-bar

Welcome to my `ProgressBar` component.

## Installation

```sh
npm i @test-easy/progress-bar
# or with yarn
yarn add @test-easy/progress-bar
```

Then to use the component in your code just import it!

```js
import ProgressBar from '@test-easy/progress-bar';
// and with css-modules
import '@test-easy/progress-bar/dist/main.css';
```
