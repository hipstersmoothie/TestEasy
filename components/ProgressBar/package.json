{
  "name": "@test-easy/progress-bar",
  "version": "0.0.0",
  "main": "./dist/cjs/index.js",
  "module": "./dist/esm/index.js",
  "types": "./dist",
  "repository": "hipstersmoothie/TestEasy",
  "author": "Andrew Lisowski lisowski54@gmail.com",
  "contributors": [
    "Andrew Lisowski lisowski54@gmail.com"
  ],
  "license": "MIT",
  "sideEffects": [
    "*.css"
  ],
  "scripts": {
    "clean": "ds clean",
    "build": "ds build",
    "dev": "ds dev",
    "storybook": "ds storybook start",
    "start": "ds build --watch",
    "test": "ds test",
    "lint": "ds lint",
    "size": "ds size"
  },
  "dependencies": {
    "@babel/runtime": "7.7.2"
  },
  "peerDependencies": {
    "react": ">= 16.8.6"
  },
  "files": [
    "dist",
    "src",
    "!*.snippet.*",
    "!*theme.*",
    "!*.test.*",
    "!__snapshots__",
    "!__tests__"
  ]
}
