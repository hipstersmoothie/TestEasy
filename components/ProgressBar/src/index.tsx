import React from 'react';
import styles from './ProgressBar.css';

interface ProgressBarProps {
  /** The value for the component to display */
  value?: string;
}

/** A simple component. */
export const ProgressBar = (props: ProgressBarProps) => {
  return (
    <div className={styles.progressBar}>
      This is a ProgressBar component {props.value}
    </div>
  );
};

export default ProgressBar;
