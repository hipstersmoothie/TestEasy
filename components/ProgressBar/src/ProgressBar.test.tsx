import React from 'react';
import { render } from '@testing-library/react';
import ProgressBar from '.';

describe('ProgressBar', () => {
  test('It matches the snapshot', () => {
    const { container } = render(<ProgressBar />);
    expect(container).toMatchSnapshot();
  });
});
