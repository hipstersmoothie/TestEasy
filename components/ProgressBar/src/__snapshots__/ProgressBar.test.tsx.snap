// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar It matches the snapshot 1`] = `
<div>
  <div
    class="progressBar"
  >
    This is a ProgressBar component 
  </div>
</div>
`;
